KeeYees ESP-WROOM-32

theZ3r0CooL opened the pull request:
- Adds KeeYees ESP-WROOM-32
+ Needs some more additions to the SVG but the necessary components are there as well as commented GPIO labels, however these are on the bottom of the board for space reasons so might just remove them.

urish reviewed (changes requested):
Thanks!

It looks like the board needs to be scaled up a bit. The distance between pins has to be 2.54mm (0.1") exactly, otherwise, it breaks the grid. Currently the distance is 2.41mm. 

Also, how did you come up with the id for the board, `keeyees-esp-wroom-32-38p`?   
I see that the fqbn says "uPesy_wroom", so perhaps something like `upesy-esp32-wroom` will make more sense as a name?

theZ3r0CooL commented:
> Thanks!
> 
> It looks like the board needs to be scaled up a bit. The distance between pins has to be 2.54mm (0.1") exactly, otherwise, it breaks the grid. Currently the distance is 2.41mm.
> 
> Also, how did you come up with the id for the board, `keeyees-esp-wroom-32-38p`? I see that the fqbn says "uPesy_wroom", so perhaps something like `upesy-esp32-wroom` will make more sense as a name?

I did initially have the GPIO spacing at 2.54, but when I tested it on the wokwi website, they didn't line up with the SVG so I adjusted them until they did. I think the issue is padding I added to the viewBox to better view the svg while working on it may mean it renders smaller than actual on the site and me adjusting the values to fit due to that probably threw it off.

I think it's a clone of the uPesy-wroom, but it has 38pins instead of 32. As far as I know, no boards in the espressif esp32 repo have 38 pins defined, but the upset--esp32-wroom is the closest. I may define it myself for my local install if I can't find a real board definition for it somewhere and I've just use the upesy board in the IDE for now.
